nipype/algorithms/misc.py

```diff
@@ -822,7 +822,7 @@ def _run_interface(self, runtime):
         df = pd.DataFrame([input_dict])
 
         if self._have_lock:
-            self._lock = pl.FileLock(self.inputs.in_file)
+            self._lock = SoftFileLock(self.inputs.in_file + '.lock')
```

Comment:
```suggestion
            self._lock = SoftFileLock('%s.lock' % self.inputs.in_file)
```

Just in case in_file is a pathlike.